Cargo.toml

```diff
@@ -65,6 +65,8 @@ dirs-next = { version = "2.0.0", optional = true }
 grep-cli = { version = "0.1.6", optional = true }
 regex = { version = "1.0", optional = true }
 walkdir = { version = "2.0", optional = true }
+time = { version = "0.3.5", features = ["formatting"] }
+bytesize = {version = "1.1.0", features = ["serde"]}
```

Comment:
Again, thanks a lot for working on this.

I wrote a proposal in #1701 that maybe we should keep using `--style`.

Another note: I'm a big believer in small and incremental steps when developing software. I think a good first minimal scope would be to only add support for showing the file size in bytes (so we don't need to add a dependency on `bytesize` in the first iteration). That way we get a clear picture of what the infrastructure would look like for supporting more kinds of info in the header.

Once that basic PR has landed, we can open new PRs that adds more header info types and e.g. adds a dependency on `bytesize`.

In my experience an incremental approach to develop software is much more pleasant for everyone. It's pleasant for maintainers since CR becomes easier. It's pleasant for contributors since they don't end up with a giant branch that gets conflicts over and over. And it's pleasant for users who gets new features quicker.

I want to stress that this is just a proposal. If you strongly prefer to do it all in one go in one PR then we can make that work too :)

Reply:
@Enselic I'll experiment with your proposal on #1701 and we'll see how it turns out, I think it's a good idea!

Regarding small incremental steps, it all comes down the trade-off between overhead of multiple PRs vs. overhead of a larger PR: it's subjective and case-by-case can differ whether a PR is large enough to justify multiple PRs, or if it's okay to stay as one as adding more PRs might cause even more overhead: for maintainers to keep track and review multiple pieces, for the contributor to manage multiple branches, and for users to have their features changed in quick succession (e.g. output is byte now, and next week it's human formatted, that's a surprise).

In this case, I personally think the addition of human formatting, last modified at and permissions doesn't add much more complexity to the PR, the majority of the code is the infrastructure, and these 3 are some basic usages of that infrastructure, on the other hand, adding file owner and group would add a lot more complexity, hence I dropped it from this PR and said we can do it later. All in all though, I'm happy to split this into smaller PRs, or continue as it is, not a big difference for me.